SocialCareCaseViewerApi.Tests/V1/IntegrationTests/UpdateWorkerDetails.cs

```diff
@@ -0,0 +1,140 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using SocialCareCaseViewerApi.V1.Boundary.Requests;
+using SocialCareCaseViewerApi.V1.Boundary.Response;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace SocialCareCaseViewerApi.Tests.V1.IntegrationTests
+{
+    [TestFixture]
+    public class UpdateWorkerDetails : IntegrationTestSetup<Startup>
+    {
+        private SocialCareCaseViewerApi.V1.Infrastructure.Worker _existingDbWorker;
+        private SocialCareCaseViewerApi.V1.Infrastructure.Worker _allocationWorker;
+        private SocialCareCaseViewerApi.V1.Infrastructure.Team _existingDbTeam;
+        private SocialCareCaseViewerApi.V1.Infrastructure.Team _differentDbTeam;
+        private SocialCareCaseViewerApi.V1.Infrastructure.Person _resident;
+
+        [SetUp]
+        public void Setup()
+        {
+            // Clear test database of any rows in the database
+            DatabaseContext.Database.ExecuteSqlRaw("DELETE from dbo.sccv_team;");
+            DatabaseContext.Database.ExecuteSqlRaw("DELETE from dbo.sccv_worker;");
+            DatabaseContext.Database.ExecuteSqlRaw("DELETE from dbo.sccv_workerteam;");
+            DatabaseContext.Database.ExecuteSqlRaw("DELETE from dbo.dm_persons;");
+            DatabaseContext.Database.ExecuteSqlRaw("DELETE from dbo.sccv_allocations;");
+
+
+            // Create existing workers,teams and worker teams
+            (var existingDbWorker, var existingDbTeam, var existingDbWorkerTeam) = IntegrationTestHelpers.SetupExistingWorker(DatabaseContext);
+            (var allocationWorker, var allocatorTeam, var allocatorWorkerTeam) = IntegrationTestHelpers.SetupExistingWorker(DatabaseContext);
+
+            var differentDbTeam = IntegrationTestHelpers.CreateAnotherTeam(DatabaseContext, existingDbWorker.ContextFlag);
+
+            // Create an existing resident that shares the same age context as existingDbWorker
+            var resident = IntegrationTestHelpers.CreateExistingPerson(DatabaseContext, ageContext: existingDbWorker.ContextFlag);
+
+            _existingDbWorker = existingDbWorker;
+            _existingDbTeam = existingDbTeam;
+            _allocationWorker = allocationWorker;
+            _differentDbTeam = differentDbTeam;
+            _resident = resident;
+        }
+
+
+        [Test]
+        public async Task UpdateWorkerWithNewTeamReturnsTheOnlyTheUpdatedTeam()
```

Comment:
If we're testing the updating of a worker I think it can be useful to have assertions before the patch. To me it proves more strongly that the PATCH request did change the team the worker belongs to.

Reply:
I agree (and I can try this again) but it came back to entity framework already knowing about the worker before it changed and led to the false positive. I'm not sure if it's the test setup or if there are ways to set entity framework to not do some of this hard to understand magic 😅 but yeah, I think I tried getting the worker first and checking their team but then entity framework did its magic because now it knew and was tracking it (which is my best guess for why the test returned a false positive when you get the worker first. 😅 

But happy to check this again